Assets/_BForBoss/_Core/Scripts/UserInterface/EntityCounterViewBehaviour.cs

```diff
@@ -0,0 +1,27 @@
+using Perigon.Entities;
+using TMPro;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class EntityCounterViewBehaviour : MonoBehaviour
+    {
+        [SerializeField] private TMP_Text _entityCounterLabel = null;
+
+        public void Initialize(LifeCycleManager lifeCycleManager)
+        {
+            lifeCycleManager.OnLivingEntityEliminated += HandleLivingEntitiesAmountChanged;
+        }
+
+        public void Reset()
+        {
+            _entityCounterLabel.text = "0";
+        }
+
+        private void HandleLivingEntitiesAmountChanged(int amount)
+        {
+            _entityCounterLabel.text = amount.ToString();
+        }
+        
```

Comment:
u should unsubscribe 
`ifeCycleManager.OnLivingEntityEliminated += HandleLivingEntitiesAmountChanged;`
in the deinit

Reply:
by Deinit do you `OnDestroy`?

Reply:
yuh